### PR TITLE
:sparkles: fix-login-page-using-new-2fa-style

### LINKED ISSUE #473: As user I want a secure admin
*Splitted from #323 *

* [x] Add 2FA to admin (You can use our default project as example since it was added there after we started this project.)
* [x] Enforce strong passwords in admin (Django has some settings I think)
* [x] Password reset needs to be present (should also be in default project).

### PR DESCRIPTION
**Pull request by azertyValentin**
Fixes #473 and replaces #477

fix-login-page-using-new-2fa-style

### CODE REVIEW

File: src/openforms/conf/base.py [hunk 1]
```diff
@@ -794,3 +794,6 @@
 OPENFORMS_LOCATION_CLIENT = config(
     "OPENFORMS_LOCATION_CLIENT", "openforms.contrib.bag.client.BAGClient"
 )
+
+
+ADMIN_DISPLAY_DROP_DOWN_MENU_CONDITION_FUNCTION = "openforms.utils.django-two-factors.should_display_dropdown_menu"
```
Comment (sergei-maertens) [inline src/openforms/conf/base.py:799]: Please group this setting with the other admin index setting(s).

Also, in admin index, the prefix should be `ADMIN_INDEX` and not just `ADMIN`, so please update that PR as well.

Comment (sergei-maertens) [reply src/openforms/conf/base.py:799]: `django-two-factors` is not a valid python identifier -> use underscores

And rather put this in `src/openforms/utils/django_two_factor_auth.py`

File: src/openforms/utils/django-two-factors.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+from django.utils.module_loading import import_string
+
+
+def should_display_dropdown_menu(request):
+    return import_string("django_admin_index.utils.should_display_dropdown_menu")(request) and request.user.is_verified()
```
Comment (sergei-maertens) [inline src/openforms/utils/django-two-factors.py:5]: This code is barely readable and there's no reason to do dynamic imports here

```suggestion
from django_admin_index.utils import should_display_dropdown_menu as default_should_display_dropdown_menu

def should_display_dropdown_menu(request) -> bool:
    return default_should_display_dropdown_menu(request) and request.user.is_verified()
```

**Review by sergei-maertens — CHANGES_REQUESTED** [review]
I'm missing the dependency update for django-admin-index, which has the `ADMIN_DISPLAY_DROP_DOWN_MENU_CONDITION_FUNCTION` setting

File: requirements/base.txt [hunk 3]
```diff
@@ -189,7 +189,8 @@ lxml==4.6.3
     # via
     #   -r requirements/base.in
     #   xmlsec
-maykin-django-two-factor-auth[phonenumbers]==2.0.2
+#maykin-django-two-factor-auth[phonenumbers]==2.0.2
+git+https://github.com/maykinmedia/django-two-factor-auth/@task-334-fix-login-page-style#egg=django-two-factor-auth
```
Comment (sergei-maertens) [inline requirements/base.txt:193]: Never update `base.txt` directly, you should be using the `pip-compile` tooling. See the header of this file:

```
#
# This file is autogenerated by pip-compile
# To update, run:
#
#    ./bin/compile_dependencies.sh
#
```

**Review by sergei-maertens — CHANGES_REQUESTED** [review]

File: src/openforms/conf/base.py (hunk 1 above)
Comment (azertyValentin) [reply src/openforms/conf/base.py:799]: done

File: src/openforms/utils/django-two-factors.py (hunk 2 above)
Comment (azertyValentin) [reply src/openforms/utils/django-two-factors.py:5]: done

File: requirements/base.txt (hunk 3 above)
Comment (azertyValentin) [reply requirements/base.txt:193]: done

**Review by sergei-maertens — APPROVED** [review]

**Comment by sergei-maertens** [comment]
Closing in favour of #568

**Pull request closed without merging**